doc/release-notes-0.4.0.rst

```diff
@@ -0,0 +1,274 @@
+* PCA63519 (Smart Remote 3 DK add-on)
+
+
+Required tools
+**************
+
+In addition to the tools mentioned in :ref:`gs_installing`, the following tool versions are required to work with the |NCS|:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Tool
+     - Version
+     - Download link
+   * - SEGGER J-Link
+     - V6.40
+     - `J-Link Software and Documentation Pack`_
+   * - nRF5x Command Line Tools
+     - v9.8.1
+     - `nRF5x Command Line Tools`_
+   * - dtc (Linux only)
+     - v1.4.6 or later
+     - :ref:`gs_installing_tools_linux`
+
+
+As IDE, we recommend to use |SES| (Nordic Edition), version 4.16 or later.
+It is available from the following links:
+
+* `SEGGER Embedded Studio (Nordic Edition) - Windows x86`_
+* `SEGGER Embedded Studio (Nordic Edition) - Windows x64`_
+* `SEGGER Embedded Studio (Nordic Edition) - Mac OS x64`_
+* `SEGGER Embedded Studio (Nordic Edition) - Linux x86`_
+* `SEGGER Embedded Studio (Nordic Edition) - Linux x64`_
+
+
+Change log
+**********
+
+The following sections provide detailed lists of changes by component.
+
+nRF9160
+=======
+
+* Added the following samples:
+
+  * :ref:`http_application_update_sample` - demonstrates a basic FOTA (firmware over-the-air) update
+  * :ref:`mqtt_simple_sample` - connects to an MQTT broker and sends and receives data
+  * :ref:`nrf_coap_client_sample` - receives data from a public CoAP server
+  * Simple GPS sample - gets the current position and logs it on UART
+
+* Renamed the :ref:`secure_partition_manager` sample from Secure Boot to Secure Partition Manager
+* Various fixes and updates to the :ref:`nrfxlib:bsdlib` (in nrfxlib):
+
+  * Updated the library to version 0.3.0 (see the :ref:`nrfxlib:bsdlib_changelog` for details)
+  * Added support for GNSS supporting GPS as a socket
+  * Implemented :cpp:func:`bsd_os_timedwait`, allowing a proper poll operation and blocking sockets
+  * Minor fixes to the nRF91 sockets SHIM layer
+
+
+Common libraries
+================
+
+* Added the following libraries:
+
+  * :ref:`lib_download_client` - downloads files over HTTP and reports back the progress (as data fragments) to the application
+  * Nordic CoAP library - ported to the |NCS|
+  * :ref:`lib_spm` (replacing Secure Boot) - configures security attributions for the flash, SRAM, and peripherals
+
+
+Crypto
+======
+
+* Added :ref:`nrfxlib:nrf_oberon_readme` v3.0.0 (see :ref:`nrfxlib:crypto_changelog_oberon` for details)
+* Added :ref:`nrfxlib:nrf_cc310_mbedcrypto_readme` v0.7.0 (initial, experimental release) to do hardware-accelerated cryptography using Arm CryptoCell CC310 in select architectures
+* Added :ref:`nrfxlib:nrf_security` v0.7.0 (initial, experimental release) to bridge software-implemented mbed TLS and hardware-accelerated alternatives through the nrf_cc310_mbedcrypto library
+* Added a glue layer to the :ref:`nrfxlib:nrf_security` to enable both mbed TLS APIs (software implementation) and hardware-accelerated cryptography at the same time
```

Comment:
Should 116 and 117 be two different items? @frkv

Reply:
Rephrased, combining the two items:
```suggestion
* Added :ref:`nrfxlib:nrf_security` v0.7.0 (initial, experimental release) to bridge software-implemented mbed TLS and hardware-accelerated alternatives, nrf_cc310_mbedcrypto library, through the use of a glue layer
```